ExceptionInInitializerError: Caused by: java.lang.UnsatisfiedLinkError: Unable to load library 'glean_ffi':

```
    FATAL EXCEPTION: GleanAPIPool
Process: org.mozilla.samples.browser, PID: 6717
java.lang.ExceptionInInitializerError
	at mozilla.telemetry.glean.GleanInternalAPI$initialize$1.invokeSuspend(Glean.kt:151)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.IllegalStateException: Glean functionality not available
	at mozilla.telemetry.glean.rust.LibGleanFFI$Companion$INSTANCE$1.invoke(LibGleanFFI.kt:79)
	at mozilla.telemetry.glean.rust.LibGleanFFI$Companion$INSTANCE$1.invoke(LibGleanFFI.kt:67)
	at java.lang.reflect.Proxy.invoke(Proxy.java:397)
	at $Proxy2.glean_new_ping_type(Unknown Source)
	at mozilla.telemetry.glean.private.PingType.<init>(PingType.kt:54)
	at mozilla.telemetry.glean.GleanMetrics.Pings.<clinit>(Pings.kt:52)
	... 10 more
Caused by: java.lang.UnsatisfiedLinkError: Unable to load library 'glean_ffi':
dlopen failed: cannot locate symbol "stderr" referenced by "libglean_ffi.so"...
dlopen failed: cannot locate symbol "stderr" referenced by "libglean_ffi.so"...
dlopen failed: cannot locate symbol "stderr" referenced by "libglean_ffi.so"...
Native library (android-x86/libglean_ffi.so) not found in resource path (.)
	at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:302)
	at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:455)
	at com.sun.jna.Library$Handler.<init>(Library.java:179)
	at com.sun.jna.Native.load(Native.java:596)
	at com.sun.jna.Native.load(Native.java:570)
	at mozilla.telemetry.glean.rust.LibGleanFFI$Companion.<clinit>(LibGleanFFI.kt:71)
	at mozilla.telemetry.glean.rust.LibGleanFFI.<clinit>(LibGleanFFI.kt)
	... 12 more
	Suppressed: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "stderr" referenced by "libglean_ffi.so"...
		at com.sun.jna.Native.open(Native Method)
		at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:191)
		... 18 more
	Suppressed: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "stderr" referenced by "libglean_ffi.so"...
		at com.sun.jna.Native.open(Native Method)
		at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:204)
		... 18 more
	Suppressed: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "stderr" referenced by "libglean_ffi.so"...
		at java.lang.Runtime.loadLibrary(Runtime.java:371)
		at java.lang.System.loadLibrary(System.java:989)
		at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:218)
		... 18 more
	Suppressed: java.io.IOException: Native library (android-x86/libglean_ffi.so) not found in resource path (.)
		at com.sun.jna.Native.extractFromResourcePath(Native.java:1089)
		at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:276)
		... 18 more
```

--- a/buildSrc/src/main/java/Dependencies.kt
+++ b/buildSrc/src/main/java/Dependencies.kt
@@ -29,7 +29,7 @@ object Versions {
 
     const val mozilla_appservices = "61.0.6"
 
-    const val mozilla_glean = "31.1.1"
+    const val mozilla_glean = "31.2.2"
 
     const val material = "1.1.0"
     const val nearby = "17.0.0"

--- a/components/service/glean/src/test/java/mozilla/components/service/glean/net/ConceptFetchHttpUploaderTest.kt
+++ b/components/service/glean/src/test/java/mozilla/components/service/glean/net/ConceptFetchHttpUploaderTest.kt
@@ -39,9 +39,7 @@ import org.mockito.Mockito.verify
 class ConceptFetchHttpUploaderTest {
     private val testPath: String = "/some/random/path/not/important"
     private val testPing: String = "{ 'ping': 'test' }"
-    private val testDefaultConfig = Configuration().copy(
-        userAgent = "Glean/Test 25.0.2"
-    )
+    private val testDefaultConfig = Configuration()
 
     /**
      * Create a mock webserver that accepts all requests.
@@ -219,7 +217,6 @@ class ConceptFetchHttpUploaderTest {
         val server = getMockWebServer()
 
         val testConfig = testDefaultConfig.copy(
-            userAgent = "Telemetry/42.23",
             serverEndpoint = "http://localhost:" + server.port
         )
 

--- a/docs/changelog.md
+++ b/docs/changelog.md
@@ -38,6 +38,16 @@ permalink: /changelog/
 * **feature-privatemode**
   * Add `PrivateNotificationFeature` to display a notification when private sessions are open.
 
+* **service-glean**
+  * Glean was updated to v31.2.2
+    * BUGFIX: Correctly format the date and time in the Date header
+    * Feature: Add rate limiting capabilities to the upload manager
+    * BUGFIX: baseline pings with reason "dirty startup" are no longer sent if Glean did not full initialize in the previous run
+    * BUGFIX: Compile dependencies with `NDEBUG` to avoid linking unavailable symbols.
+      This fixes a crash due to a missing `stderr` symbol on older Android.
+
+
+
 # 47.0.0
 
 * [Commits](https://github.com/mozilla-mobile/android-components/compare/v46.0.0...v47.0.0)

